galley/pkg/server/controller.go

```diff
@@ -29,37 +29,19 @@ const (
 	DefaultReadinessProbeFilePath = "/healthReadiness"
 )
 
-var (
-	livenessProbeController, readinessProbeController probe.Controller
-	galleyProbe                                       *probe.Probe
-)
-
 //StartProbeCheck start probe check for Galley
-func StartProbeCheck(livenessProbeOptions *probe.Options, readinessProbeOptions *probe.Options, stop <-chan struct{}) {
+func StartProbeCheck(livenessProbeController *probe.Controller, readinessProbeController *probe.Controller, stop <-chan struct{}) {
 	defer func() {
-		galleyProbe.SetAvailable(nil)
 		if livenessProbeController != nil {
-			livenessProbeController.Close()
+			(*livenessProbeController).Close()
```

Comment:
The extra parenthesis are not needed here, line 39, and lines 43-44 if controller interface is not passed via a pointer. `livenessProbeController.Close()` should be sufficient.

Reply:
done